.github/workflows/ci.yml

```
@@ -3,44 +3,70 @@ name: CI
 on: pull_request
 
 jobs:
-  tests:
-
-    name: Test Python ${{ matrix.python_version }}
+  lint:
+    name: Lint
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        python_version: ['3.6', '3.7', '3.8', '3.9']
-    env:
-      PYTHONDEVMODE: 1
 
     steps:
     - uses: actions/checkout@v2
 
     - name: Set up Python
       uses: actions/setup-python@v2
       with:
-        python-version: ${{ matrix.python_version }}
+        python-version: '3.x'
+        cache: 'pip'
+        cache-dependency-path: 'pyproject.toml'
 
-    - name: Poetry setup
+    - name: Install Flit
       run: |
-        python -m pip install poetry
+        python -m pip install flit
 
     - name: Install dependencies
       run: |
-        poetry install
+        flit install --extras dev
```

Don't we want to be explicit about our Python environment? I suppose that is quite pedantic though...

```suggestion
        python -m flit install --extras dev
```